oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoVersionGCSupport.java

```diff
@@ -136,7 +136,7 @@ protected SplitDocumentCleanUp createCleanUp(Set<SplitDocType> gcTypes,
             // makes any future similar problem more visible than long running
             // queries alone (15min is still long).
             Iterable<NodeDocument> iterable = filter(transform(getNodeCollection().find(query)
-                    .maxTime(15, TimeUnit.MINUTES),
+                    .maxTime(15, TimeUnit.MINUTES).hintString("_sdType_1__sdMaxRevTime_1"),
```

Comment:
This name looks somewhat cryptic and generated by MongoDB. Is this something we can rely on? What happens when there is no such index? Will the query fail or MongoDB just silently ignore the hint? The reason why I'm asking is because there are cases where Oak does not automatically created that index. See https://github.com/apache/jackrabbit-oak/blob/72203f3928226a26ea22197aed9c8901605618f2/oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/mongo/MongoDocumentStore.java#L354

I'm also wondering whether it is possible to read the name of the index from MongoDB or use the other method variant `hint(Bson)`. I was hoping the latter does not depend on how MongoDB generates a name for the index.

Reply:
comments incorporated